public/components/metrics/top_menu/top_menu.tsx

```diff
@@ -141,16 +145,35 @@ export const TopMenu = ({
 
   const handleSavingObjects = async () => {
     let savedMetricIds = [];
+    const start = 'now-15y';
+    const end = 'now';
+    const span = 1;
+    const resolution = 'hours';
 
     try {
       savedMetricIds = await Promise.all(
         sortedMetricsLayout.map(async (metricLayout, index) => {
+          console.log('visualizationsMetaData: ', visualizationsMetaData);
           const updatedMetric = updateMetricsWithSelections(
             visualizationsMetaData[index],
-            dateSpanFilter.start,
-            dateSpanFilter.end,
-            dateSpanFilter.span + dateSpanFilter.resolution
+            // dateSpanFilter.start,
+            // dateSpanFilter.end,
+            // dateSpanFilter.span + dateSpanFilter.resolution
+            start,
+            end,
+            span + resolution
           );
+          console.log('metricLayout: ', metricLayout);
+
+          if (metricLayout.metricType === 'openTelemetryMetric') {
+            const updatedOpenTelemetryMetric = updateOpenTelemetryMetricsWithSelections(
```

Comment:
All this code is in MetricExport now.  We'll want to consider how to take away the conditional branching here.  Probably we'll end up with a couple classes that will greatly help how this works.

Reply:
yup changed it to MetricExport